src/Eshopworld.Messaging/TopicAdapter.cs

```diff
@@ -124,5 +130,19 @@ protected override void Dispose(bool disposing)
 
             base.Dispose(disposing);
         }
+
+        /// <inheritdoc />
+        protected override void RebuildReceiver()
+        {
+            Receiver?.CloseAsync().Wait();
+            Receiver = new MessageReceiver(ConnectionString, EntityNameHelper.FormatSubscriptionPath(AzureTopic.Name, SubscriptionName), ReceiveMode.PeekLock, new RetryExponential(TimeSpan.FromMilliseconds(100), TimeSpan.FromMilliseconds(500), 3), BatchSize);
+        }
+
+        /// <inheritdoc />
+        protected override void RebuildSender()
+        {
+            Sender?.CloseAsync().Wait();

```

Comment:
should we perhaps wrap with try catch?

Reply:
That try catch code was there from previous v2 client integrations.
The v3 client ultimately does this:
https://github.com/Azure/azure-amqp/blob/master/Microsoft.Azure.Amqp/Singleton.cs#L49-L54

Which is a call to Dispose on the AMQP client (other implementations are similar), and that won't throw.